scripts/uberenv/uberenv.py

```diff
@@ -625,6 +625,7 @@ def install(self):
                       return res
         # note: this assumes package extends python when +python
         # this may fail general cases
+        full_spec = self.read_spack_full_spec(self.pkg_name,self.opts["spec"])
```

Comment:
Thanks @adammoody .  This was actually fixed in the main uberenv repo, which unfortunately we have diverged from due to a very long standing PR on that repo.  Unfortunately there are probably 4-5 uberenvs around at this point.

Can you remove line 628 (what you added) and actually move line 612 right above 609 so we don't diverge even further?